sycamore/transforms/regex_replace.py

```diff
@@ -0,0 +1,62 @@
+import re
+
+from ray.data import Dataset
+
+from sycamore.data import Document
+from sycamore.plan_nodes import Node, Transform, SingleThreadUser, NonGPUUser
+from sycamore.transforms.map import generate_map_function
+
+COALESCE_WHITESPACE = [
+    (r"\s+", " "),
+    (r"^ ", ""),
+    (r" $", ""),
+]
+
+
+class RegexReplace(SingleThreadUser, NonGPUUser, Transform):
+    """
+    The RegexReplace transform modifies the text_representation in each
+    Element in every Document.
+
+    Args:
+        child: The source node or component that provides the documents
+        spec: A list of tuples of regular expressions and substitutions,
+              to be executed in order via re.sub()
+        kwargs: Additional resource-related arguments that can be passed to the operation
+
+    Example:
+        .. code-block:: python
+            rr = RegexReplace(child=node, spec=[(r"\s+", " "), (r"^ ", "")])
```

Comment:
Since you don't need the kwargs and don't use them in examples, remove them here?